2.scDINO_processing/temporal_shiny_app/server.r

```diff
@@ -0,0 +1,31 @@
+# Load necessary libraries
+library(shiny)
+library(ggplot2)
+
+
+# Define server logic
+server <- function(input, output, session) {
+    umap_df <- read.csv("CLS_features_annotated_umap.csv")
+    # Create well select input
+    # Create well checkbox group input
+    output$doseSelect <- renderUI({
+    checkboxGroupInput("Doses", "Select Doses:", choices = unique(umap_df$Metadata_dose), selected = unique(umap_df$Metadata_dose))
+    })
+
+    # Create time checkbox group input
+    output$timeSelect <- renderUI({
+    checkboxGroupInput("times", "Select Time Points:", choices = unique(umap_df$Metadata_Time), selected = unique(umap_df$Metadata_Time))
+    })
+
+    output$umapPlot <- renderPlot({
+    # Filter data based on selected wells and time points
+    data <- umap_df[umap_df$Metadata_dose %in% input$Doses & umap_df$Metadata_Time %in% input$times, ]
```

Comment:
upgrading to `dplyr::filter()` will help with slow loading times, FWIW

Reply:
This helped!